File: backend/src/main/java/com/bidbridge/backend/entity/Notification.java

package com.bidbridge.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;
import java.time.Instant;

@Data
@NoArgsConstructor
@Entity
@Table(name = "notifications", schema = "public")
public class Notification {

    @Id
    @GeneratedValue
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    private String type;

    @Column(columnDefinition = "TEXT")
    private String data;

    private Boolean isRead = false;
    private Instant createdAt;
}

File: frontend/src/types/notification.ts
export interface Notification {
  id: string;
  profile_id: string;
  type: string;
  data: string;
  is_read: boolean;
  created_at: string;
}

export interface NotificationApiResponse {
  id: string;
  profileId: string;
  type: string;
  data: string;
  isRead: boolean;
  createdAt: string;
}

export function transformNotificationFromApi(
  apiNotification: NotificationApiResponse
): Notification {
  return {
    id: apiNotification.id,
    profile_id: apiNotification.profileId,
    type: apiNotification.type,
    data: apiNotification.data,
    is_read: apiNotification.isRead,
    created_at: apiNotification.createdAt,
  };
}
